drivers/firmware/scmi/nxp/bbm.c

```diff
@@ -67,3 +74,33 @@ int scmi_bbm_button_event(uint32_t *flags)
 
 	return 0;
 }
+
+static void scmi_bbm_event_protocol_cb(struct scmi_channel *channel, struct scmi_message msg)
+{
+	uint32_t flags;
+	int32_t ret;
+	uint32_t msg_id = SCMI_MESSAGE_HDR_EX_MSGID(msg.hdr);
+
+	if (msg_id == SCMI_PROTO_BBM_PROTOCOL_BUTTON_EVENT) {
+		ret = scmi_bbm_button_event(&flags);
+		if (ret < 0) {
+			LOG_ERR("failed to read bbm button event to shmem: %d", ret);
+		} else {
+			printf("SCMI BBM BUTTON notification: flags=0x%08X\n", flags);
+		}
+	}
+}
+
+static struct scmi_protocol_event bbm_event = {
+	.id = SCMI_PROTOCOL_NXP_BBM,
+	.evts = scmi_nxp_bbm_events,
+	.num_events = ARRAY_SIZE(scmi_nxp_bbm_events),
+	.cb = scmi_bbm_event_protocol_cb,
+};
+
+static int scmi_nxp_bbm_event_init(void)
+{
+	return scmi_register_protocol_event_handler(&bbm_event);
```

Comment:
do we need to also unregister protocol events? if not, then we can probably get around with binding the event array to the protocol structure at build time.